scipy/conftest.py

```diff
@@ -208,20 +239,18 @@ def skip_or_xfail_xp_backends(xp, backends, kwargs, skip_or_xfail='skip'):
     Parameters
     ----------
     *backends : tuple
-        Backends to skip, e.g. ``("array_api_strict", "torch")``.
+        Backends to skip, e.g. ``("array_api_strict", "numpy")``.
         These are overriden when ``np_only`` is ``True``, and are not
         necessary to provide for non-CPU backends when ``cpu_only`` is ``True``.
```

Comment:
I think this needs to change - `*backends` can only be empty or of length 1 now. Maybe better to just turn the parameter into accepting either a `backend` or defaulting to `None`?

Reply:
Not quite, actually.
This is the docstring of `skip_or_xfail_xp_backends`, and it does take a tuple of backend names (not an unpacked tuple, so fixed in the last commit).

The user-facing `skip_xp_backends` does accept a single backend, yes --- through the cryptic `request` argument.
Then this line, `backends, kwargs = _backends_kwargs_from_request(request, skip_or_xfail='skip')` combines this single backend from `request` with all other backends from previous skips, and returns a tuple, e.g. `(`array_api_strict`, `torch`)`. 

Spooky action on a distance, via pytest internals :-).

Reply:
IOW, shortened up and annotated:

```
@pytest.fixture
def skip_xp_backends(xp, request):
    # ...
    backends, kwargs = _backends_kwargs_from_request(request, skip_or_xfail='skip')   # <<<< here
    skip_or_xfail_xp_backends(xp, backends, kwargs, skip_or_xfail='skip')


def _backends_kwargs_from_request(request, skip_or_xfail):
    """A helper for {skip,xfail}_xp_backends"""
    # do not allow multiple backends
    args_ = request.keywords[f'{skip_or_xfail}_xp_backends'].args     # get _this_ @skip_xp_backends argument
    if len(args_) > 1:
        # ....   args_ is `"(numpy,)"`
        raise ValueError(f"multiple backends: {args_}")

    markers = list(request.node.iter_markers(f'{skip_or_xfail}_xp_backends'))   # combine!

    backends = []
    kwargs = {}
    for marker in markers:
         # ... process
    return backends, kwargs

```

Reply:
Okay yeah, I see. Since the APIs of the two have now diverged, could you either write a docstring or add the info to the narrative docs? Right now there is nothing to explicitly tell people that, for example, they must only provide one backend at a time.

Reply:
OK, added short docstrings.